Cython/Compiler/Main.py

```diff
@@ -66,7 +68,7 @@ def __init__(self, include_directories, compiler_directives, cpp=False,
         self.modules = {"__builtin__" : Builtin.builtin_scope}
         self.cython_scope = CythonScope.create_cython_scope(self)
         self.modules["cython"] = self.cython_scope
-        self.include_directories = include_directories
+        self.include_directories = tuple(include_directories)
```

Comment:
Does this really need to be a tuple? It seems like it should rather be a list, at least at this point, but also generally. Make a copy when necessary, but why use a tuple?

Reply:
ok, changing to list

Reply:
it has to be a tuple to be hashable. Reverting

Reply:
Where do you need it to be hashable?

Reply:
`search_include_directories` is marked as `Utils.cached_function` on line 606
```
@Utils.cached_function
def search_include_directories(dirs, qualified_name, suffix, pos, include=False):
```

Reply:
Then it's probably enough to convert it into a tuple right before passing it into the function, right?
I mean, the right solution might actually be to convert everything to tuples inside of `cached_function()`, but in any case, we shouldn't use surprising data structures in multiple places just because one place that uses them requires a certain type as a side-effect of an optimisation.

Reply:
Would you prefer `args = [tuple(x) for x in args if isinstance(x, list) else x]` in `cached_function`? I reduced the "multiple places to the call to `search_include_directories` and added a comment.